### ISSUE TITLE
MAX will Not Run - White Screen then Exits

### ISSUE THREAD

**Issue by ArgentArts** [TheGameCreators/GameGuruRepo#1526]
I have had MAX uninstalled for months. Just downloaded the updater from my account today and installed MAX. There were no previous installs of MAX on my hard drives. After successfully installing MAX, I tried to run it. MAX starts, shows the titlebar at the top with "GameGuru MAX" on it, the rest of the screen is white. Then MAX exits after about a second. There are no error messages displayed.

My system specs are: intel i9-10900K, 64GB RAM, Nvidia 1080ti (latest game-ready drivers), Windows 10 (all updates). 

From what I can tell, I have all redistributables installed and the .Net 3.0 framework (latest). 

I've looked in the MAX folder and I don't see any log file generated.

**Comment by rvanner** [comment]
@LeeBamberTGC & @plemsoft Seems like a show stopper for this user. Can you both look into this ASAP?

**Comment by plemsoft** [comment]
@ArgentArts Thanks for reporting this issue, if you do not see the splash screen but only get a white screen, it sound like a very early problem. this could be a missing/bad shader ... could you try to run "GameGuru MAX Updater.exe" from your Max folder and select "Verify Files". Just to make sure you dont have any bad/missing files.
Also if you have any special antivirus software , try to disable it while you install , just to make sure your antivirus is not blocking any files. Thanks for your help :)

**Comment by ArgentArts** [comment]
I've already attempted to verify the files and the installer is telling me that everything is okay. The only antivirus I am running is Windows Defender and nothing else.

Other things I've tried:

- Running the GameGuruMAX.exe as an administrator
- Running the launcher as an administrator
- Rolling back to several previous versions
- Uninstalling and ensuring there are no traces of GameGuruMAX or GameGuruAPPs, etc. and reinstalling fresh, etc.

**Comment by ArgentArts** [comment]
BTW - I don't appear to be the only one getting this issue. Here's a thread on the GameGuru forum where someone reported the same thing back at the end of September:

https://forum.game-guru.com/thread/223044

**Comment by Gusher** [comment]
Which CPU are you using? I remember awhile back on another game engine with similiar issues that a certain intel cpu model has issues, but not on AMD..

**Comment by rvanner** [comment]
@LeeBamberTGC & @plemsoft Can you work with @ArgentArts to debug what's happening on his PC? This is a showstopper issue.

**Comment by LeeBamberTGC** [comment]
@rvanner All I can think is that it relates to the audio issue we saw on your PC, but something similar but not identical. If the crash happens at the very start, it's either Wicked shaders out of date or some sound API conflict.  Any hope of testing on another computer to see if it relates to hardware or some soft-related reason.

**Comment by ArgentArts** [comment]
@LeeBamberTGC - Unfortunately, I do not have another PC I can test this on. What else can I do to help identify what's going on? I'm not the only one who's experienced this error (see my link posted above for another MAX user who has experienced this). That would indicate that, once released, there may be more people who would experience this as well. And that would lead to bad reviews for MAX, I would think. 

BTW - I had installed and was able to run MAX many months back. So, it did run on my system at one time.

Is there any more information I can provide you that would help? Would you like me to create and post a video of what happens when I try to run MAX?

**Comment by LeeBamberTGC** [comment]
@ArgentArts Yes a video would be great as it can often show subtle signs so I know where to focus. Also send me any "C:\Users\XXX\Documents\GameGuruApps\GameGuruMAX\Guru-MapEditor.log" file if one gets created.

**Comment by ArgentArts** [comment]
Here's the video:

http://www.monkeyfrogstudio.com/uploads/max_fail.mp4

One thing to note - sometimes MAX exits sooner (quits faster) and sometimes it hangs on for just a bit longer. 

As far as the .log file, there isn't a "GameGuruApps" folder in my Documents folder.

**Comment by LeeBamberTGC** [comment]
@ArgentArts Would it be okay if I send you a debug build of the executable. It will give us source code and line number of the crash, which will tell us exactly what is happening?

**Comment by ArgentArts** [comment]
Definitely. I would be happy to do whatever it takes to sort this out.

**Comment by LeeBamberTGC** [comment]
@ArgentArts Here is a DEBUG build of the latest MAX. It probably will not run, but when it crashes immediately it should show you a dialog. If you can snapshot that dialog, I can learn more:

https://drive.google.com/file/d/14WrHMqFeGwsvFqLI2f69zqlzjTPpTp5l/view?usp=sharing

**Comment by ArgentArts** [comment]
![max_dialog_01](https://user-images.githubusercontent.com/36735507/141399487-b819f9b9-cba7-4e50-a9d8-bbefcb3c8800.png)

**Comment by LeeBamberTGC** [comment]
@ArgentArts This relates to the DXCOMPILER.DLL failing to create one of its objects:

				HRESULT hr = DxcCreateInstance(CLSID_DxcUtils, IID_PPV_ARGS(&dxcUtils));
				assert(SUCCEEDED(hr));

In translation, this means that we are using a particular version of DXCOMPILER.DLL that expects some DirectX environment that would provide us with an instance of CLSID_DxcUtils.  Possible things to try, check you have the latest DirectX redists (they may have been corrupted). Double-check you are definitely 64-bit operating system and that your GPU supports DirectX 12.

@plemsoft @PaulSJ Any other ideas that we can try quickly to go through a process of elimination before we dive deeper into trying different DXCOMPILER versions at our end. The latest is from July 2021 on MSs GitHub site.

**Comment by ArgentArts** [comment]
Since I'd reported this (first post), my PC has been updated to Windows 11, so it's definitely running DirectX as that's a requirement to be able to update. Just in case, I've run dxdiag and here's what it says:

![direcx12](https://user-images.githubusercontent.com/36735507/141418044-180c2f5a-ca14-4d55-9118-a526ecb7b3c8.png)

And definitely 64-bit. I've been running a 64-bit OS for many years now. ;)

**Comment by plemsoft** [comment]
@ArgentArts If you click next page in dxdiag , under drivers -> feature levels do you see 11_1,11_0 ?

**Comment by ArgentArts** [comment]
@plemsoft :

![dx_02](https://user-images.githubusercontent.com/36735507/141429254-d2592fbc-800d-4e30-adf9-dcc366121da8.png)

**Comment by plemsoft** [comment]
@ArgentArts Please try this version and let me know the last message you get:

https://drive.google.com/file/d/1-whlcZ4d-5b82khphRBBskulCXrUeADW/view?usp=sharing

Thanks for your help :)

**Comment by GraPhiX-Guru** [comment]
@ArgentArts  can i ask what refresh rate and screen resolution you are using ?

**Comment by ArgentArts** [comment]
@GraPhiX-Guru - 3440 x 1440 at 100Hz.

**Comment by ArgentArts** [comment]
@plemsoft - This one actually runs on my system. After various messages, I am able to get to the storyboard section and then, via exiting the storyboard, to the main level editing GUI. I am able to move about the main viewport, too.

**Comment by LeeBamberTGC** [comment]
@plemsoft What was the magic sauce?  Looks like you forged a path through!

**Comment by ArgentArts** [comment]
So, will this fix be included in the Friday update? Or should I wait until next week's release before I update?

**Comment by LeeBamberTGC** [comment]
@plemsoft If Preben is happy the fix is careful, and keeps stability for our Friday release, and he sees this tonight, I will add it :)

**Comment by plemsoft** [comment]
@LeeBamberTGC Sorry i think i cant make it today, it will require some new shaders, and more testing but i know the problem and will explain on TW , @ArgentArts Sorry don't think i can make it for today, but for sure in the next update. Anyway i will post a version tomorrow that you can use until then :)

**Comment by ArgentArts** [comment]
No worries and no hurry (on my part).

**Comment by ArgentArts** [comment]
@plemsoft Should I not try using this version to learn MAX? When I place a player start and hit the rocket to start the level, I get the following error:

![runtime](https://user-images.githubusercontent.com/36735507/141528132-86961470-9a78-49ef-a22b-52578acf472e.png)

**Comment by plemsoft** [comment]
@ArgentArts Yes but that file is part of the new Friday build , so this is what you should do.

1. Install the latest Friday build.
2. Download the new .exe from below link.
3. Install it into the latest Friday build.

And you should be good to go.
Let me know if everything works for you, and i will get the commit to Lee :)

https://drive.google.com/file/d/10Bzj8Xm7UaL_TIYPFMMtDRpB6RFoM0fG/view?usp=sharing

Thanks for your help.

**Comment by ArgentArts** [comment]
Thanks. Worked perfectly.

**Comment by LeeBamberTGC** [comment]
This will be in the next build :)

**Comment by ArgentArts** [comment]
Awesome! Thanks! :)

**Comment by dllllllllllb** [comment]
@ArgentArts Hi, could you confirm if the issue is now fixed for you :]
